packages/discord.js/src/util/Constants.js

```diff
@@ -65,6 +65,13 @@ exports.NonSystemMessageTypes = [
  * @typedef {DMChannel|TextChannel|NewsChannel|ThreadChannel} TextBasedChannels
  */
 
+/**
+ * Data that resolves to give a text-based channel. This can be:
+ * * A text-based channel
+ * * A snowflake
+ * @typedef {TextBasedChannels|Snowflake} TextBasedChannelsResolvable
```

Comment:
the grammar in this sounds... odd. Can't this be TextBasedChannelResolvable? I know the other typedef is called TextBasedChannels but maybe that should be updated too because afaik we usually go with the singular for these typedefs

Reply:
[TextBasedChannel](https://github.com/discordjs/discord.js/blob/b162f27e46524bfc64515969d753c6e8f30e6c40/packages/discord.js/src/structures/interfaces/TextBasedChannel.js) already exists and is a mixin, that's why the typedef is in plural.